roles/kubernetes-apps/ingress_controller/ingress_nginx/templates/ds-ingress-nginx-controller.yml.j2

```diff
@@ -29,6 +29,12 @@ spec:
       nodeSelector:
         {{ ingress_nginx_nodeselector | to_nice_yaml }}
 {%- endif %}
+{% if ingress_nginx_tolerations %}
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
```

Comment:
key should be `ingress_nginx_nodeselector` and not hardcoded

Reply:
The issue exists. You hardcode the `node-role.kubernetes.io/master` as key here. But the `nodeSelector` right above is a variable that can be set `{{ ingress_nginx_nodeselector | to_nice_yaml }}`

Reply:
If I use to_nice_yaml, indent will be invalid.

Then  I can compromise following code:

```
{% if ingress_nginx_tolerations_key %}
      tolerations:
        - effect: NoSchedule
          key: {{ ingress_nginx_tolerations_key }}
{% endif %}
```

Is it fine for you? If not, could you tell me concrete code?